### PR TITLE
Tile source metadata

### PR DESCRIPTION
**Pull request by chinmaychahar**
Metadata for examples/models/tile_source.py

- [x] issues: #11765

### CODE REVIEW

**Comment by chinmaychahar** [comment]
@bryevdv One thing I struggle with every time is to find the correct metadata for the `ref` section. I wanted some guidance for the same. 

For example, for this file i.e. tile_source.py the reference URL will be [this](https://docs.bokeh.org/en/latest/docs/user_guide/topics/geo.html#tile-provider-maps) so do I add the permalink to the main heading for the `ref` section here?

**Review by bryevdv — APPROVED** [review]

File: examples/models/tile_source.py [hunk 1]
```diff
@@ -1,3 +1,11 @@
+''' Example to demonstrate creating map-based visualizations and 
+working with geographical data using WMTSTileSource in Bokeh. 
+
+.. bokeh-example-metadata::
+    :apis: bokeh.models.WMTSTileSource, bokeh.models.Range1d, bokeh.models.BoxZoomTool, bokeh.models.PanTool, bokeh.models.WheelZoomTool
+    :refs: :ref:`ug_topics_tile_provider_maps`
+    :keywords: tile source
+'''
```
Comment (bryevdv) [inline examples/models/tile_source.py]: Formatting consistency with other examples (blank line after metadata)
```suggestion
    :keywords: tile source
    
'''
```

**Review by bryevdv — CHANGES_REQUESTED** [review]
One small comment from me @chinmaychahar but also some linter issues:
```
E         File contains trailing whitespace: examples/models/tile_source.py, line 1.
E         File contains trailing whitespace: examples/models/tile_source.py, line 2.
```
I do recommend running the codebase checks before pushing as all these issues can be caught locally

**Comment by bryevdv** [comment]
Thanks @chinmaychahar !

**Pull request merged**